vitest.config.ts

```diff
@@ -18,6 +19,7 @@ const includeWithStyles = [
 const isReact17 = version.startsWith("17");
 
 export default defineConfig({
+  plugins: [react()],
```

Comment:
How did this work without the `react` plugin before? We might also need to make this Vitest config more generic, as we want to use Vitest (and the same test files) to test Solid examples too.

The test files don't include React or JSX. The components are imported for each test in a separate `vitest.setup.ts` file.

Reply:
¯\_(ツ)_/¯. It didn't work for me without it. Potentially vite/vitest was doing something magic with the previous tsconfigs?

Reply:
I can confirm that it doesn't work without the plugin installed. Not sure what change affected this, but I guess it's fine for now. I wonder if it'll complicate things when adding tests to the Solid port, but that's for another day.

Reply:
Here's an example test error. All tests fail, probably with the same error. For some reason I'm only seeing the error for the first failing test, not sure why 🤷 

<img width="470" alt="image" src="https://github.com/user-attachments/assets/1c66ff89-a9cc-4802-8be4-88e17567e190" />

Reply:
I've tested on a clean `main` workspace, and the change that causes this is:

```diff
-"jsx": "react-jsx",
+"jsx": "preserve",
```

Reply:
I think you'll want this and want to have separate test configs using Vitest's [workspace configurations](https://vitest.dev/guide/workspace.html) to have multiple targets with different plugins so that you can have both react and solid.

Reply:
That makes sense to me. I think we should try it once we implement Solid tests (there are none at the moment).